### ISSUE TITLE
Adding class to <div id="app"> leads to missing html replacement without error

### ISSUE THREAD

**Issue by cimchd** [antfu/vite-ssg#118]
We had an issue with missing replaced html content after the ssg build. In our team someone has added a css class to the div in the index.html:

```html
<html>
  <!-- ... -->
  <body>
    <div id="app" class="h-full"></div>
    <script type="module" src="/src/main.ts"></script>
  </body>
</html>
```
This caused vite-ssg to skip the replacement of the div due to this code:
https://github.com/antfu/vite-ssg/blob/e8dd90a7de1af8e5b23f65904394761e95cce5f7/src/node/build.ts#L189-L193

It took us a while to find this error. Maybe it is better to add a replacement regex that is more tolerant, something like:
```js
 return indexHTML
    .replace(
     /<div ([^>]*)id="app"\s?([^>]*)>\s*<\/div>/,
      (m, g1, g2) => 
         `<div ${g1}id="app" ${g2 ? g2+' ': ''}data-server-rendered="true">${appHTML}</div>${stateScript}`
    )
```
Or a check if `<div id="app"></div>` could be found in the index.html and trigger a warning if not. Or maybe even both.

If you are interested in one of the two solutions, I could provide a PR.

**Comment by wtho** [comment]
Just happened to stumble on the very same problem 🤔 

I'd suggest you  just provide a PR with the regex and see what Anthony will say.

**Comment by cimchd** [comment]
closed by #183
